imap_processing/ccsds/excel_to_xtce.py

```diff
@@ -363,6 +366,34 @@ def _add_analog_conversion(self, row: pd.Series, encoding: Et.Element) -> None:
                 term.attrib["coefficient"] = str(conversion[col])
                 term.attrib["exponent"] = str(i)
 
+    def _add_state_conversion(self, row: pd.Series, parameter_type: Et.Element) -> None:
+        """
+        Add a state conversion to the parameter type.
+
+        Changing from an IntegerParameterType to an EnumeratedParameterType. Adding
+        the list of state mappings to the parameter type.
+
+        Parameters
+        ----------
+        row : pandas.Row
+            Row to be added to the XTCE file, containing mnemonic, packetName.
+        parameter_type : Element
+            The parameter type element to add the conversion to.
+        """
+        # It is an EnumeratedParameterType rather than an IntegerParameterType
+        parameter_type.tag = "xtce:EnumeratedParameterType"
+        enumeration_list = Et.SubElement(parameter_type, "xtce:EnumerationList")
+        # Lookup the enumeration states for this parameter from the States sheet
+        state_sheet = self.sheets["States"]
+        state_sheet = state_sheet.loc[
+            (state_sheet["packetName"] == row["packetName"])
+            & (state_sheet["mnemonic"] == row["mnemonic"])
+        ]
+        for _, state_row in state_sheet.iterrows():
+            enumeration = Et.SubElement(enumeration_list, "xtce:Enumeration")
+            enumeration.attrib["value"] = str(state_row["value"])
+            enumeration.attrib["label"] = str(state_row["state"])
```

Comment:
I am wondering if this function produces something like this?
```
	        <xtce:EnumeratedParameterType name="QUARTER_CYCLE_ENUM" signed="false">
                <xtce:IntegerDataEncoding sizeInBits="5" encoding="unsigned"/>
                <xtce:EnumerationList>
                    <xtce:Enumeration label="FIRST" value="0"/>
                    <xtce:Enumeration label="SECOND" value="1"/>
		   <xtce:Enumeration label="THIRD" value="2"/>
                    <xtce:Enumeration label="FORTH" value="3"/>
                </xtce:EnumerationList>
            </xtce:EnumeratedParameterType>
```

I saw most of it but didn't see it add this line. I think we need that.
```
                <xtce:IntegerDataEncoding sizeInBits="5" encoding="unsigned"/>
```

Reply:
That is taken care of in the above `_add_parameter_type()`, which adds the encodings. Then here we modify that parameter type to be `EnumeratedParameterType` **instead of** `IntegerParameterType`. It should be there if you run this on the swapi or swe definitions (could you verify it works for you as well). This is where tests would be nice...

Reply:
I ran locally and it looks correct.